include/swift/AST/ClangTypeConverter.h

```diff
@@ -78,9 +78,10 @@ class ClangTypeConverter :
   /// declaration introduced by this converter, and if so, return the original
   /// Swift declaration.
   Decl *getSwiftDeclForExportedClangDecl(const clang::Decl *decl) const;
+      
+  clang::QualType convert(Type type);
```

Comment:
If I'm not mistaken, this is an implementation detail and should not be exposed. @varungandhi-apple?

Reply:
Yeah, I'm a bit wary of this. Could we instead have a method on `ClangTypeConverter` which handles translating a templated C++ function type to a Swift function type, given a substitution map? I would've understood if this method needed to be used in a lot of places, but it only seems to have two external clients in this patch.

Reply:
`ASTContext::Implementation` already has a `ClangTypeConverter` member so, I think it makes sense to put this helper function there.

Reply:
Sorry I missed this comment somehow.

> did you mean to have a method that translates a templated C++ function type to a Swift function type on ClangModuleLoader? That sounds like a place that would keep ClangTypeConverter encapsulated and allow us to do what we need in this PR.

No, the type conversion methods are all on `ClangTypeConverter`. As Zoe said, the ASTContext carries around a `ClangTypeConverter` so that can be used (it is already being used in a handful of places). `ClangModuleLoader` shouldn't have methods for converting types.

Reply:
👍